lib/active_interaction/filter.rb

```diff
@@ -205,12 +209,14 @@ def describe(value)
       "(Object doesn't support #inspect)"
     end
 
+    # @param interaction [Base, nil]
+    #
     # @return [Object]
-    def raw_default
+    def raw_default(interaction)
       value = options.fetch(:default)
 
       if value.is_a?(Proc)
-        value.call
+        interaction.instance_exec(&value) if interaction
```

Comment:
`eagerly_evaluate_default` calls it with `nil` because it happens before the instance exists. There is no explicit test, but lots of tests exercise this functionality.